applications/StructuralMechanicsApplication/custom_elements/updated_lagrangian.cpp

```diff
@@ -466,13 +492,61 @@ void UpdatedLagrangian::CalculateOnIntegrationPoints(
         for ( IndexType point_number = 0; point_number < mConstitutiveLawVector.size(); ++point_number )
             rValues[point_number] = mDetF0[point_number];
     } else {
+        const bool f_computed = mF0Computed;
+        if (rCurrentProcessInfo[STEP] > 1) mF0Computed = false;
         BaseSolidElement::CalculateOnIntegrationPoints(rVariable, rValues, rCurrentProcessInfo);
+        if (rCurrentProcessInfo[STEP] > 1) mF0Computed = f_computed;
     }
 }
 
 /***********************************************************************************/
 /***********************************************************************************/
 
+void UpdatedLagrangian::CalculateOnIntegrationPoints(
+    const Variable<array_1d<double, 3>>& rVariable,
+    std::vector<array_1d<double, 3>>& rValues,
+    const ProcessInfo& rCurrentProcessInfo
+    )
+{
+    const bool f_computed = mF0Computed;
+    if (rCurrentProcessInfo[STEP] > 1) mF0Computed = false;
+    BaseSolidElement::CalculateOnIntegrationPoints(rVariable, rValues, rCurrentProcessInfo);
+    if (rCurrentProcessInfo[STEP] > 1) mF0Computed = f_computed;
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void UpdatedLagrangian::CalculateOnIntegrationPoints(
+    const Variable<array_1d<double, 6>>& rVariable,
+    std::vector<array_1d<double, 6>>& rValues,
+    const ProcessInfo& rCurrentProcessInfo
+    )
+{
+    const bool f_computed = mF0Computed;
+    if (rCurrentProcessInfo[STEP] > 1) mF0Computed = false;
+    BaseSolidElement::CalculateOnIntegrationPoints(rVariable, rValues, rCurrentProcessInfo);
+    if (rCurrentProcessInfo[STEP] > 1) mF0Computed = f_computed;
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void UpdatedLagrangian::CalculateOnIntegrationPoints(
+    const Variable<Vector>& rVariable,
+    std::vector<Vector>& rValues,
+    const ProcessInfo& rCurrentProcessInfo
+    )
+{
+    const bool f_computed = mF0Computed;
+    if (rCurrentProcessInfo[STEP] > 1) mF0Computed = false;
```

Comment:
just to say that this would not work if the method would be `const` (as it should be IMO)

Reply:
Well...certainly I have been thinking another way to fix this, but I cannot as the flag mF0Computed is reseted in InitializeSOlutionStep, but the postprocess is executed after FinalizeSolutionStep...

Reply:
Maybe, instead of bool I can use an enum or an int, and that way I can retrieve more info

Reply:
...